### ISSUE TITLE
Virtual navigation property throws exception after updating to v0.28.0

### ISSUE THREAD

**Issue by rioka** [hazzik/DelegateDecompiler#162]
I was about to do some changes in an old application, initially built in .NET 4.5.2, and using DelegateDecompiler v0.18.1.

This is a simple scenario to replicate the issue
- Two types (`Order` and `OrderLine`), 1-M relation between them
- Navigation property from parent to child was defined using [encapsulated collection](https://lostechies.com/jimmybogard/2014/05/09/missing-ef-feature-workarounds-encapsulated-collections/)

A minimal sample solution to replicate this issue is available [here](https://github.com/rioka/DelegateCompilerSample)

### LSS
Because of some complex dependency chain, I eventually decided to update DelegateDecompiler to latest version (v0.28.0).

When I run `DelegateCompilerSample.Use0_18_1.exe`, everything works as expected.
When I run `DelegateCompilerSample.Use0_28_0.exe`, I now get an exception.

```
Unhandled Exception: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.ArgumentNullException: Value cannot be null.
Parameter name: key
   at System.Collections.Generic.Dictionary`2.FindEntry(TKey key)
   at System.Collections.Generic.Dictionary`2.TryGetValue(TKey key, TValue& value)
   at DelegateDecompiler.MethodBodyDecompiler.ReplaceMethodCallsExpressionVisitor.VisitMember(MemberExpression node)
   at System.Linq.Expressions.MemberExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at System.Linq.Expressions.ExpressionVisitor.VisitUnary(UnaryExpression node)
   at System.Linq.Expressions.UnaryExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at DelegateDecompiler.MethodBodyDecompiler.DecompileVirtual(Type declaringType, MethodInfo method, IList`1 args)
   at DelegateDecompiler.MethodBodyDecompiler.Decompile(MethodInfo method, Type declaringType)
   at DelegateDecompiler.DecompileExtensions.<>c__DisplayClass6_0.<.cctor>b__1()
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.LazyInitValue()
   at System.Lazy`1.get_Value()
   at DelegateDecompiler.DecompileExtensions.Decompile(MethodInfo method, Type declaringType)
   at DelegateDecompiler.DecompileExpressionVisitor.Decompile(MethodInfo method, Expression instance, IList`1 arguments)
   at DelegateDecompiler.DecompileExpressionVisitor.VisitMember(MemberExpression node)
   at System.Linq.Expressions.MemberExpression.Accept(ExpressionVisitor visitor)
   at System.Linq.Expressions.ExpressionVisitor.Visit(Expression node)
   at DelegateDecompiler.DecompileExpressionVisitor.Decompile(Expression expression)
   at DelegateCompilerSample.EF.EntityTypeConfigurationEx.HasMany[TEntityType,TTargetEntity](EntityTypeConfiguration`1 c, Expression`1 navigationPropertyExpression)
...
```

- There is a new `HasMany` extension method (`EntityTypeConfigurationEx.cs`), which accepts an IEnumerable instead of an `ICollection` (IIRC I grabbed this code from jbogard).

From what I can see, the exception is caused by [this line](https://github.com/hazzik/DelegateDecompiler/blob/d134b819243a253301ebfbc3e797fbfde191954f/src/DelegateDecompiler/MethodBodyDecompiler.cs#L175) in `MethodBodyDecompiler.cs`, as `GetGetMethod` returns `null`.

If navigation property (`Order.OrderLines`, `Order._lines`) is not `virtual`, it works with both versions.

Is there anything I can change in my code to get the "old" behavior?
Or is this simply an unsupported scenario now?

Thanks

**Comment by hazzik** [comment]
Fixed in 0.28.2

**Comment by rioka** [comment]
kudos
Thanks a lot
